File: apps/web/src/types/alerts.ts
export type AlertSummary = {
  id: string;
  intel_file_id: string;
  intel_file_title: string;
  alert_type: string;
  severity: string;
  message: string;
  status: string;
  created_at: string;
};

export type AlertStatusFilter = "all" | "pending" | "sent" | "acknowledged" | "dismissed";

export type AlertSeverityFilter = "all" | "info" | "watch" | "important" | "urgent";

export type AlertListData = {
  items: AlertSummary[];
  total: number;
};

export type AlertUpdateRequest = {
  status: "acknowledged" | "dismissed";
};

File: backend/app/schemas/alerts.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import AlertSeverity, AlertStatus, AlertType


class AlertUpdateRequest(BaseModel):
    status: AlertStatus


class AlertSummary(BaseModel):
    id: UUID
    intel_file_id: UUID
    intel_file_title: str
    alert_type: AlertType
    severity: AlertSeverity
    message: str
    status: AlertStatus
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AlertListData(BaseModel):
    items: list[AlertSummary]
    total: int
